src/vs/workbench/contrib/terminal/browser/terminalProfileQuickpick.ts

```diff
@@ -197,20 +203,45 @@ export class TerminalProfileQuickpick {
 		if (!result) {
 			return undefined;
 		}
+		if (!await this._isProfileSafe(result.profile)) {
+			return undefined;
+		}
 		if (keyMods) {
 			result.keyMods = keyMods;
 		}
 		return result;
 	}
 
+	private async _isProfileSafe(profile: ITerminalProfile | IExtensionTerminalProfile): Promise<boolean> {
+		if (!('isUnsafePath' in profile)) {
+			return true;
+		}
+
+		// Get the user's explicit permission to use a potentially unsafe path
+		return await new Promise<boolean>(r => {
+			const handle = this._notificationService.prompt(
+				Severity.Warning,
+				nls.localize('unsafePathWarning', 'This profile uses the potentially unsafe path {0} that another user can write to. Are you sure you want to use it?', `"${profile.path}"`),
```

Comment:
Motivations for suggesting a slight change in phrasing:
- The image shows a weird line break in the middle of a sentence.
- Sentence ending in "to" feels awkward.

`This profile uses a potentially unsafe path that can be modified by another user: {0}. Are you use you want to use it?`